src/utils/item.ts

```diff
@@ -130,3 +130,87 @@ export const paginationContentFilter = (
   items
     .filter((i) => i.type !== ItemType.FOLDER)
     .filter((i) => !i.settings?.isPinned);
+
+interface Node {
+  children: string[];
+}
+
+interface Tree {
+  [nodeId: string]: Node;
+}
+
+const createMapTree = (data: DiscriminatedItem[]) => {
+  const childrenTreeMap: Tree = {};
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < data.length; i++) {
+    const ele = data[i];
+    const parents = getParentsIdsFromPath(ele.path, { ignoreSelf: true });
+    const lastParent = parents[parents.length - 1];
+    if (lastParent) {
+      if (childrenTreeMap[lastParent]) {
+        childrenTreeMap[lastParent].children = [
+          ...childrenTreeMap[lastParent].children,
+          ele.id,
+        ];
+      } else {
+        childrenTreeMap[lastParent] = { children: [ele.id] };
+      }
+    }
+  }
+  return childrenTreeMap;
+};
+
+type ItemWithChildren = DiscriminatedItem & {
+  children: ItemWithChildren[];
+};
+
+interface TreeNode {
+  [nodeId: string]: ItemWithChildren;
+}
+
+// handle item children tree
+const buildItemsTree = (data: DiscriminatedItem[]) => {
+  const tree: TreeNode = {};
+  if (data.length === 1) {
+    tree[data[0].id] = { ...data[0], children: [] };
+  }
```

Comment:
Do you need a special case for 1? Won't it work with the normal process?

Reply:
What about this?

Reply:
basically, we build our tree based on parent-children relation which will not work for a non-children Item, As we build a map tree based on parent